src/server/routes/api/index.js

```diff
@@ -114,31 +111,87 @@ export function handleSqsDataProcessing(sqsClient, type) {
   };
 }
 
+/**
+* constructDateQuery(dateInput, isEndDate = false)
+* @desc Builds the queries of the dates for sending to the API
+*/
+function constructDateQuery(dateInput, isEndDate = false) {
+  const lastSecond = (isEndDate) ? 'T23:59:59' : '';
+
+  if (dateInput && typeof dateInput === 'string') {
+    const dateArray = dateInput.split('/');
```

Comment:
Again, not a big deal, but you could do this with a regex:

```
date_matches = dateInput.match(/^(\d{2})\/(\d{2})\/((?:\d{2})|(?:\d{4}))$/)
if (!date_matches) {
  return undefined;
}
```
Then you should still check if the month and day is valid.

Saves a few lines at least!

Reply:
This is a nice approach!

Reply:
Thanks for letting me know! I have implement the regex in the codes and pushed it.